specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,289 @@
+  ///         Storage
+  ///     * Cookies 
+  ///   * HTTP Cache 
+  ///   * Download History
+  ///   * General Autofill 
+  ///   * Password Autosave
+  ///   * Browsing History
+  ///   * Settings  
+  /// The completed handler will be invoked when the browsing data has been cleared and will 
+  /// indicate if the specified data was properly cleared.
+  /// If the WebView object is closed before the clear browsing data operation 
+  /// has completed, the handler will be released, but not invoked. In this case 
+  /// the clear browsing data operation may or may not be completed. 
+  /// ClearBrowsingData clears the `dataKinds` for all time. 
+  HRESULT ClearBrowsingData(
+      [in] UINT64 dataKinds,
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+  
+  /// ClearBrowsingDataInTimeRange takes in two additional parameters for the 
+  /// start and end time for which it should clear the data between.  The startTime and endTime 
+  /// parameters correspond to the number of seconds since the UNIX epoch. 
+  ///
+  /// \snippet AppWindow.cpp ClearBrowsingData
+  HRESULT ClearBrowsingDataInTimeRange(
+      [in] UINT64 dataKinds, 
+      [in] double startTime,
+      [in] double endTime, 
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+}
+
+/// The caller implements this interface to receive the ClearBrowsingData result.
+[uuid(27676699-FE17-4E2B-8C1B-267395A04ED5), object, pointer_default(unique)]
+interface ICoreWebView2ClearBrowsingDataCompletedHandler : IUnknown {
+  /// Provide the completion status and result of the corresponding
+  /// asynchronous method. The result indicates if the ClearBrowsingData 
+  /// call has completed successfully and all intended data has been cleared. 
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] BOOL isSuccessful);
+}
+
+```
+### .NET, WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    [Flags] enum CoreWebView2BrowsingDataKind
+    {
+        AppCache = 1,
+        FileSystems = 2,
+        IndexedDb = 4,
+        LocalStorage = 8,
+        WebSql = 16,
+        CacheStorage = 32,
+        DomStorage = 63,
+        Cookies = 64,
+        Site = 127,
+        HttpCache = 128,
+        DownloadHistory = 256,
+        GeneralAutofill = 512,
+        PasswordAutosave = 1024,
+        BrowsingHistory = 2048,
+        Settings = 4096,
+        Profile = 8191,
+    };
+
+    public partial class CoreWebView2Environment
+    {
+        /// The ClearBrowsingDataAsync method may be overloaded to take either one parameter - dataKinds, or three parameters - dataKinds, startTime, and endTime.
+        /// NOTE TO REVIEWERS:
+        /// We're taking ulong here because we need to accept orred enum values like HttpCache | DownloadHistory which isn't a value in the DataKind enum. 
+        /// We need to take ulong to get the proper masked thru, is it OK to take the enum as the parameter type here in COM, WinRT, and .NET?
```

Comment:
It seems to work just fine in C# & C++. It seems reasonable that it would work since the point of flags enums is that you can `or` the values together. We can leave the question here but can we go with the enum as the type instead of ulong/UINT64 for both this and the COM IDL? Does anything fail if we do that?

Reply:
Actually changing this to use the flags enum allows dotnet/winrt to use the enum so changing it to that.